HW05.Rmd

```diff
@@ -0,0 +1,158 @@
+
+#According to the graph, "haematopoietic_and_lymphoid_tissue" has the lowest AUC values to the compound vorinostat.
+print(paste("Cancer", AUC_vor[1,1], "has the lowest AUC values to the compound vorinostat."))
+```
+
+
+## Which compound is the prostate cancer cell line 22RV1 most sensitive to? (For 22RV1, which compound has the lowest AUC value?)
+
+### It looks there is no observation associated with cancer cell line 22RV1 in the data set. So, I used all prostate cancer records instead. 
+
+```{r graph_2, echo=TRUE, message=FALSE, warning=FALSE}
+#limit data to prostate cancer
+AUC_22RV1 <- AUC_Compound %>%
+  filter(cancer_type == "prostate" & is.na(cpd_name) == FALSE & is.na(area_under_curve) == FALSE ) %>%
+  group_by(cpd_name) %>%
+  summarize(AUC = mean(area_under_curve)) %>%
+  arrange(AUC) %>%
+  top_n(-10, AUC)
+
+#generate a bar plot to show the compound with the lowest AUC
+ggplot(AUC_22RV1, aes(x = reorder(cpd_name, AUC), y = AUC, fill = cpd_name)) + 
+  geom_col() + 
+  scale_y_continuous(breaks = seq(0, 8, by = 1)) +
+  labs(title = "AUC by Compound for Prostate Cancer", x = "\nCompound Name", y = "AUC\n") +
+  theme(plot.title = element_text(hjust = 0.5, size = 12, face = "bold"), axis.text.x = element_text(angle = 45, hjust=1, size=8), axis.title.x = element_text(size = 10, face = "bold"), axis.title.y = element_text(size = 10, face = "bold"))
+
+# According to the graph, compound "ouabain" has the lowest AUC value. 
+print(paste("Compund", AUC_22RV1[1,1], "has the lowest AUC value."))
+```
+
+
+## For the 10 compounds that target EGFR, which of them has (on average) the lowest AUC values in the breast cancer cell lines? 
+
+```{r graph_3, echo=TRUE, message=FALSE, warning=FALSE}
+#limit data to compounds that target EGFR and in the breast cancer cell lines
+AUC_EGFR <- AUC_Compound %>%
+  filter(str_detect(gene_symbol_of_protein_target, "EGFR") & cancer_type == "breast" & is.na(area_under_curve) == FALSE ) %>%
+  group_by(cpd_name) %>%
+  summarize(AUC = mean(area_under_curve)) %>%
+  arrange(AUC) 
+
+#generate a bar plot to show the compound with the lowest AUC
+ggplot(AUC_EGFR, aes(x = reorder(cpd_name, AUC), y = AUC, fill = cpd_name)) + 
+  geom_col() + 
+  scale_y_continuous(breaks = seq(0, 15, by = 3)) +
+  labs(title = "AUC by Compound Targeting EGFR for Breast Cancer", x = "\nCompound Name", y = "AUC\n") +
+  theme(plot.title = element_text(hjust = 0.5, size = 12, face = "bold"), axis.text.x = element_text(angle = 45, hjust=1, size=8), axis.title.x = element_text(size = 10, face = "bold"), axis.title.y = element_text(size = 10, face = "bold"))
+
+# According to the graph, compound "afatinib" has the lowest AUC value. 
+print(paste("Compund", AUC_EGFR[1,1], "has the lowest AUC values in the breast cancer cell lines."))
+```
```

Comment:
After reviewing your homework, I realized that I should also include titles for my plots to make my homework easier to follow.